src/feditest/protocols/__init__.py

```diff
@@ -22,19 +22,25 @@ class Node(ABC):
     so FediTest can control and observe what it needs to when attempting to
     participate with the respective protocol.
     """
-    def __init__(self, rolename: str, node_driver: 'NodeDriver') -> None:
+    def __init__(self, rolename: str, hostname: str, node_driver: 'NodeDriver') -> None:
         """
         rolename: name of the role in the constellation
+        hostname: the hostname of this Node
         node_driver: the NodeDriver that provisioned this Node
         """
         self._rolename = rolename
+        self._hostname = hostname
         self._node_driver = node_driver
 
 
     def rolename(self):
         return self._rolename
 
 
+    def hostname(self):
```

Comment:
Just curious if there's a reason to define these methods instead of using self.hostname (for example). Even if it's a "public" field for now, if you want to make it a method later you can use an `@property` decorator to make it a method without changing the callers.

Reply:
Old habit of trying to prevent accidental changes from the outside of the object. (There may be better ways to do this in Python?)

Reply:
I understand. The `@property` decorator is often used for this purpose.

```python
@property
def hostname(self):
  return self._hostname
```

It's still "read only", but can be accessed like a field/property instead of a function. But, it's fine either way.